util/git/client.go

```diff
@@ -365,14 +366,28 @@ func (m *nativeGitClient) Fetch(revision string) error {
 }
 
 // LsFiles lists the local working tree, including only files that are under source control
-func (m *nativeGitClient) LsFiles(path string) ([]string, error) {
-	out, err := m.runCmd("ls-files", "--full-name", "-z", "--", path)
-	if err != nil {
-		return nil, err
+func (m *nativeGitClient) LsFiles(path string, enableNewGitFileGlobbing bool) ([]string, error) {
+	if enableNewGitFileGlobbing {
+		// This is the new way with safer globbing
+		err := os.Chdir(m.root)
```

Comment:
Is this sufficient to prevent directory enumeration outside the repo dir?

Reply:
That's a very good remark. I have added code below to filter out returned files that would be coming from outside of the repo, either because of a suitably crafted search pattern or because of a symlink. I know AppSets do not currently follow symlinks, but in case they do in the future (which I hope) then we won't have a hidden security issue. When a file is filtered out from the results a warning is logged.

I have also added unit tests for all that. The old `git ls-files` method will return unresolved symlinks. I didn't changed that. One more unit test checks the new method does not return symlinks that resolve outside of the repo. One last unit test checks files outside of the repo can't be returned using a specifically crafted search pattern.